api/orders.js

```diff
@@ -95,6 +95,51 @@ function getOrders(request, response, next) {
     return promise.spread(getAccountOrders);
   };
 
+  function getParsedOrders(offers) {
+    return _.reduce(offers, function(orders, off) {
+      var taker_gets;
+      var taker_pays;
+
+      var isTakerGetsXRP = _.isString(off.taker_gets);
+      var isTakerPaysXRP = _.isString(off.taker_pays);
+
+      if (isTakerGetsXRP) {
+        taker_gets = {
+          currency: 'XRP',
+          counterparty: '',
+          value: utils.dropsToXrp(off.taker_gets)
+        };
+      } else {
+        taker_gets = {
+          currency: off.taker_gets.currency,
+          counterparty: off.taker_gets.issuer,
+          value: off.taker_gets.value
+        };
+      }
```

Comment:
maybe we could refactor this to

```
taker_gets  = parseCurrencyAmount(off.taker_gets);

function parseCurrencyAmount (currencyAmount) {
    if (_.isString(currencyAmount)) {
        ...
    } else {
        ...
    }
}
```

that way we can use it elsewhere when we parse currency amount values?